src/qtt/algorithms/functions.py

```diff
@@ -337,20 +339,38 @@ def fit_gauss_ramsey(x_data, y_data, weight_power=0, maxiter=None, maxfun=5000,
 
     """
 
-    def func(params): return cost_gauss_ramsey(x_data, y_data, params, weight_power=weight_power)
+    def gauss_ramsey_model(x, amplitude, decay_time, frequency, phase, offset):
+        """  """
+        y = gauss_ramsey(x, [amplitude, decay_time, frequency, phase, offset])
+        return y
+
+    if weight_power is None:
+        weights = None
+    else:
+        diff_x = np.diff(x_data)
+        weights =  np.hstack( (diff_x[0], diff_x) )  ** weight_power
+        print(weights)
 
     if initial_params is None:
-        initial_params = estimate_parameters_damped_sine_wave(x_data, y_data, exponent=2)
+        initial_parameters = estimate_parameters_damped_sine_wave(x_data, y_data, exponent=2)
+    else:
+        initial_parameters = initial_params
+    lmfit_model = Model(gauss_ramsey_model)
+    lmfit_model.set_param_hint('amplitude', min=0)
+    lmfit_result = lmfit_model.fit(
+        y_data, x=x_data, **dict(zip(lmfit_model.param_names, initial_parameters)), verbose=verbose >= 2, weights = weights)
+
+    import qtt.algorithms.fitting
```

Comment:
Why local import instead of global
`from qtt.algorithms.fitting import extract_lmfit_parameters`
We prefer the global one, unless there is something against it.

Reply:
The global import causes a circular reference. Solving that would mean moving the method from `qtt.algorithms.functions` to `qtt.algorithms.fitting` which is not backwards compatible.